eslint-custom-rules/queries-only-in-use-query-functions.js

```diff
@@ -0,0 +1,73 @@
+module.exports = {
+  name: 'queries-only-in-use-query-functions',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'useQuery and useMutation can only be used in functions starting with use and ending with Query or Mutation respectively',
+      recommended: false,
+    },
+    messages: {
+      forbiddenUseQuery:
+        'useQuery can only be used in functions starting with use and ending with Query',
+      forbiddenUseMutation:
+        'useMutation can only be used in functions starting with use and ending with Mutation',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+
+  create(context) {
+    function isInValidFunction(node, expectedSuffix) {
+      let current = node
+      while (current) {
+        if (
+          current.type === 'FunctionDeclaration' ||
+          current.type === 'ArrowFunctionExpression' ||
+          current.type === 'FunctionExpression'
+        ) {
+          if (current.type === 'FunctionDeclaration') {
+            const functionName = current.id?.name
+            return functionName
+              ? /^use.*$/.test(functionName) && functionName.endsWith(expectedSuffix)
```

Comment:
je ne vois pas l'utiliter de la regexp

```suggestion
              ? functionName.startsWith('use') && functionName.endsWith(expectedSuffix)
```